stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/HiddenMessageListItemFilter.kt

```diff
@@ -0,0 +1,24 @@
+package io.getstream.chat.ui.sample.feature.chat
+
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.ui.messages.view.MessageListView
+import io.getstream.chat.android.ui.utils.extensions.isDeleted
+import io.getstream.chat.android.ui.utils.extensions.isGiphyEphemeral
+
+object HiddenMessageListItemFilter : MessageListView.MessageListItemFilter {
+
+    private val theirDeletedMessagePredicate: (MessageListItem) -> Boolean = { item ->
+        item is MessageListItem.MessageItem && item.message.isDeleted() && item.isTheirs
+    }
+
+    private val theirGiphyEphemeralMessagePredicate: (MessageListItem) -> Boolean = { item ->
+        item is MessageListItem.MessageItem && item.message.isGiphyEphemeral() && item.isTheirs
+    }
+
+    override fun filterMessageListItem(messageListItem: List<MessageListItem>): List<MessageListItem> {
```

Comment:
You can use `Iterable` instead of List for the parameter.

Reply:
This use case is tightened to MessageListView, so it requires to return a list and receives a list as an argument. Since it's strictly defined use-cases I prefer to keep the List type here